hs_core/views/utils.py

```diff
@@ -43,12 +43,12 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    has_edit = res.edit_users.filter(pk=user.pk).exists()
-    has_view = res.view_users.filter(pk=user.pk).exists()
-    has_full = res.owners.filter(pk=user.pk).exists()
+    has_edit = res.raccess.edit_users.filter(pk=user.pk).exists()
```

Comment:
This is not sufficient because res.raccess.edit_users does not account for res.raccess.immutable. 
Consider 
has_edit = user.uaccess.can_view_resource(res)
has_view = user.uaccess.can_change_resource(res)
has_full = user.uaccess.owns_resource(res) 
CAUTION: immutable flags apply to even owners.